torch/csrc/jit/ir.cpp

```diff
@@ -620,6 +620,42 @@ void Node::findSchema() const {
   schema_ = &getOperatorFor(this).schema();
 }
 
+namespace {
+
+const std::unordered_set<Symbol>& nondeterminstic_aten_ops() {
+  static std::unordered_set<Symbol> nondeterministic_ops = {
+    aten::dropout,
+    aten::batch_norm,
+    aten::_fused_dropout,
+    aten::_standard_gamma,
+    aten::_th_bernoulli,
+    aten::bernoulli,
+    aten::multinomial,
+    aten::normal,
+    aten::poisson,
+    aten::rrelu,
+    aten::rrelu_with_noise
+  };
+  return nondeterministic_ops;
+}
+
+}  // namespace
+
+bool Node::isNondeterministic() const {
+  if (nondeterminstic_aten_ops().count(kind()) == 0) {
+    return false;
+  }
+  // Dropout with train = False is deterministic
+  if (kind() == aten::dropout && is_constant(attr::train) && !get<bool>(attr::train).value()) {
+    return false;
+  }
+  // batch_norm with training = False is deterministic
```

Comment:
in training=False mode, it will give different results because of running mean/avg computation, right?

Reply:
Sure, but it's not really non-deterministic. I'm just saying it might be a better idea to pick a different name for this function, because we will end up using it in places where we really care about the non-determinism, but it will have this weird edge case. Think of what happened to `isSubtypeOf`.